support/sier-codec/src/builder.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{FieldDef, Object, StructDef, Type, Value};
+
+use std::collections::HashMap;
+
+pub struct Builder<'s> {
+    struct_def: &'s StructDef,
+    field_values: HashMap<String, Value>,
+}
+
+impl<'s> Builder<'s> {
+    pub(crate) fn new(struct_def: &'s StructDef) -> Self {
+        Builder {
+            struct_def,
+            field_values: HashMap::new(),
+        }
+    }
+
+    pub fn set(mut self, field_name: &str, value: impl Into<Value>) -> Self {
+        self.field_values
+            .insert(field_name.to_string(), value.into());
+        self
+    }
+
+    pub fn try_build(mut self) -> Result<Object<'s>, BuildError> {
+        let values = self
+            .struct_def
+            .fields
+            .iter()
+            .map(|field| self.value_for_field(field))
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
This could be a single call to `std::iter::fold`, couldn't it?

Reply:
I don't think so. I think this is the best way to write it as it is.

Fold is used when you need to accumulate a value across everything in an iterator. Each of these values is independent of each-other, so I don't think fold is right/necessary.